app/src/components/Premium/PremiumModal.tsx

```diff
@@ -0,0 +1,196 @@
+    setError(null);
+    if (!licenseKey.startsWith('LOR-')) {
+      setError('License Key must begin with "LOR-"');
+      return false;
+    }
+
+    return true;
+  };
+
+  const activateLicenseKey = async () => {
+    if (!checkLicenseKey()) {
+      return;
+    }
+
+    setIsActivating(true);
+
+    try {
+      const { activated, message } = await ipcRenderer.invoke(
+        'activate-license',
+        licenseKey
+      );
+
+      if (activated) {
+        setIsActivationSuccessful(true);
+      } else {
+        setError(message);
+      }
+    } catch (e: unknown) {
+      log.error(e);
+      if (e instanceof Error) {
+        setError(e.message);
+      }
+    } finally {
+      setIsActivating(false);
+    }
+  };
+
+  const closeHandler = () => {
+    if (isActivationSuccessful) {
+      activatePremium();
+    }
+
+    onClose();
+  };
+
+  return (
+    <>
+      <Modal isOpen={isOpen} onClose={closeHandler}>
+        <ModalOverlay />
+        <ModalContent>
+          {isActivationSuccessful ? (
+            <>
+              <ModalHeader>Premium Unlocked</ModalHeader>
+              <ModalCloseButton />
+              <ModalBody>
+                <div
+                  style={{
+                    display: 'flex',
+                    flexDirection: 'column',
+                    justifyContent: 'center',
+                    alignItems: 'center',
+                  }}
+                >
+                  <Confetti
+                    width={440}
+                    height={350}
+                    recycle={false}
+                    numberOfPieces={200}
+                  />
+                  <Float
+                    circleShadowDimensions={{
+                      marginTop: 20,
+                      width: 120,
+                    }}
+                  >
+                    <img
+                      src={Celebration}
+                      alt="Working hard"
+                      style={{
+                        width: '200px',
+                        border: '1px solid lightgray',
+                        borderRadius: '50%',
+                      }}
+                    />
+                  </Float>
+                  <Text
+                    style={{ marginTop: 16 }}
+                    fontWeight="medium"
+                    fontSize="lg"
+                  >
+                    Thanks for supporting Left on Read!
+                  </Text>
+                </div>
+              </ModalBody>
+              <ModalFooter>
+                <Button colorScheme="purple" onClick={onClose}>
+                  See New Analytics
+                </Button>
+              </ModalFooter>
+            </>
+          ) : (
+            <>
+              <ModalHeader>Unlock Premium</ModalHeader>
+              <ModalCloseButton />
+
+              <ModalBody>
+                <div>
+                  <Text fontWeight="medium" style={{ marginBottom: 6 }}>
+                    Your License Key
```

Comment:
I think slightly confusing language. Maybe
```suggestion
                    Input Your License Key
```